ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -0,0 +1,228 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.tools.migrations.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.execution.expression.tree.BooleanLiteral;
+import io.confluent.ksql.execution.expression.tree.CreateArrayExpression;
+import io.confluent.ksql.execution.expression.tree.CreateMapExpression;
+import io.confluent.ksql.execution.expression.tree.CreateStructExpression;
+import io.confluent.ksql.execution.expression.tree.DecimalLiteral;
+import io.confluent.ksql.execution.expression.tree.DoubleLiteral;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.IntegerLiteral;
+import io.confluent.ksql.execution.expression.tree.LongLiteral;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.metastore.TypeRegistry;
+import io.confluent.ksql.parser.AstBuilder;
+import io.confluent.ksql.parser.DefaultKsqlParser;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.tree.InsertValues;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class CommandParser {
+  private static final String QUOTED_STRING_OR_WHITESPACE = "('([^']*|(''))*')|\\s+";
+  private static final KsqlParser KSQL_PARSER = new DefaultKsqlParser();
+  private static final String INSERT = "INSERT";
+  private static final String INTO = "INTO";
+  private static final String VALUES = "VALUES";
+  private static final String SELECT = "SELECT";
+  private static final String CREATE = "CREATE";
+  private static final String SINK = "SINK";
+  private static final String SOURCE = "SOURCE";
+  private static final String CONNECTOR = "CONNECTOR";
+  private static final String SHORT_COMMENT_OPENER = "--";
+  private static final String SHORT_COMMENT_CLOSER = "\n";
+  private static final String LONG_COMMENT_OPENER = "/*";
+  private static final String LONG_COMMENT_CLOSER = "*/";
+  private static final char SINGLE_QUOTE = '\'';
+  private static final char SEMICOLON = ';';
+
+  private CommandParser() {
+  }
+
+  public static List<SqlCommand> parse(final String sql) {
+    return splitSql(sql).stream()
+        .map(CommandParser::transformToSqlCommand)
+        .collect(Collectors.toList());
+  }
+
+  /*
+  * Splits a string of sql commands into a list of commands and filters out the comments.
+  *
+  * @return a list of strings
+  * */
+  @VisibleForTesting
+  static List<String> splitSql(final String sql) {
+    final List<String> commands = new ArrayList<>();
+    String current = "";
+    for (int i = 0; i < sql.length(); i++) {
+      if (sql.charAt(i) == SINGLE_QUOTE) {
+        final int closingToken = sql.indexOf(SINGLE_QUOTE, i + 1);
```

Comment:
What happens if the sql string is invalid and doesn't contain a closing quote to go with the opening quote? Same question for malformed comments.

Reply:
Also a note for anyone else looking at this: this doesn't explicitly handle escaped quotes but that's fine since quotes are pieced back together in the final string anyway. Could be good to add this as a comment in the code itself so future readers don't wonder the same thing.

Reply:
Added a check to throw on malformed string/comments